### PR TITLE
do not load all events to check for conflicts

### PR DESCRIPTION
**Pull request by marcosh**
currently all events are loaded to check if there are concurrency conflicts.

It should be enough to load events from the initial playhead of the uncommitted events.

This could save some computations in case there are a bit of events before the conflicting playhead

### CODE REVIEW

**Review by AlessandroMinoccheri — CHANGES_REQUESTED** [review]
Is it possible to add a test for this behavior?

**Comment by marcosh** [comment]
@AlessandroMinoccheri what would you like to test exactly, since the functionality is exactly the same as before and already tested in https://github.com/broadway/broadway/blob/master/test/Broadway/EventStore/ConflictResolvingEventStoreTest.php?

**Comment by AlessandroMinoccheri** [comment]
I would like to know if it's possible to explicit well this behavior. I was mostly wondering if it was possible or not @marcosh

**Comment by marcosh** [comment]
> I would like to know if it's possible to explicit well this behavior. I was mostly wondering if it was possible or not @marcosh

@AlessandroMinoccheri the behaviour should not be changing, that's why I think already present tests should be enough. This is basically just a refactoring

**Review by AlessandroMinoccheri — APPROVED** [review]

**Comment by marcosh** [comment]
hay, anyone there? is broadway officialy a dead project?

**Comment by AlessandroMinoccheri** [comment]
What do you think @othillo of this PR? For me it's ok.

File: src/Broadway/EventStore/ConcurrencyConflictResolvingEventStore.php [hunk 1]
```diff
@@ -94,6 +96,16 @@ private function getCurrentPlayhead(DomainEventStream $committedEvents): int
         return $playhead;
     }
 
+    private function getStartingPlayhead(DomainEventStream $uncommittedEvents): int
+    {
+        $events = iterator_to_array($uncommittedEvents);
+        /** @var DomainMessage $firstEvent */
+        $firstEvent = current($events);
+        $playhead = $firstEvent->getPlayhead();
```
[the remark this answers is not in the document]
Comment (marcosh) [reply src/Broadway/EventStore/ConcurrencyConflictResolvingEventStore.php:108]: if `$uncommittedEvents` is empty, I'd find it odd to receive a `DeplicatePlayheadException`.

Anyway, I added a check to skip every work if the `DomainEventStream` is empty

[the remark this answers is not in the document]
Comment (othillo) [reply src/Broadway/EventStore/ConcurrencyConflictResolvingEventStore.php:108]: I don't mean to throw a `DuplicatePlayheadException`, I'm just saying that if `$uncommittedEvents` is empty PHP will error because the `$firstEvent->getPlayhead()` method does not exist.

Maybe we should add a test for appending zero uncommitted events to be sure.

[the remark this answers is not in the document]
Comment (marcosh) [reply src/Broadway/EventStore/ConcurrencyConflictResolvingEventStore.php:110]: I meant that this private method is called only in case of a `DuplicatePlayheadException`. And in that case I would find it odd to receive an empty `$uncommittedEvents`.

Anyway, I added the test to ensure everything works as expected.

Moreover, I added a `$id = (string)$id` to throw a `Throwable` even in case `$uncommittedEvents` is empty, to avoid creating a breaking change on the API

[the remark this answers is not in the document]
Comment (marcosh) [reply src/Broadway/EventStore/ConcurrencyConflictResolvingEventStore.php:110]: @othillo is there anything I could do on this to get this merged?

**Review by wjzijderveld — APPROVED** [review]
I'll fix the coding-standards issue in master

**Comment by wjzijderveld** [comment]
Thanks @marcosh !
